### ISSUE TITLE
I1. Instruments and Lexicons

### ISSUE THREAD

**Issue by zednis** [USGCRP/gcis-ontology#35]
Create terms for “instrument_instance” and possible for “lexicon”

examples:
- https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager
- https://data.globalchange.gov/lexicon/echo
- https://data.globalchange.gov/lexicon/ceos

Additionally, we are currently using dbpedia:Lexicon and skos:Concept for lexicons; is that an accurate approach?

**Comment by zednis** [comment]
I am not sure if this is still an issue after some of the recent changes we have made to lexicons.  For example, we are now using type `skos:ConceptScheme` for lexicon instances in RDF instead of `skos:Concept`.

@justgo129 Can you confirm that this issue is still unaddressed?

**Comment by justgo129** [comment]
Sure.  The instrument_instance issue is still unaddressed.  Please also
inform whether dbpedia:Lexicon is still an acceptable term (to couple with
a skos:ConceptScheme).  e.g.
https://data.globalchange.gov/lexicon/ceos.thtml

On Wed, Jul 1, 2015 at 3:18 PM, Stephan Zednik notifications@github.com
wrote:

> I am not sure if this is still an issue after some of the recent changes
> we have made to lexicons. For example, we are now using type
> skos:ConceptScheme for lexicon instances in RDF instead of skos:Concept.
> 
> @justgo129 https://github.com/justgo129 Can you confirm that this issue
> is still unaddressed?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/35#issuecomment-117799393
> .

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by zednis** [comment]
1) Could you rephrase the instrument_instance issue? 

2) Having looked at the RDF for dbpedia:Lexicon (see http://dbpedia.org/data/Lexicon.n3), it seems that `dbpedia:Lexicon` is not a `rdfs:Class` or `owl:Class`.  It is simply an RDF instance and not intended to be used as a type.

It is my recommendation that we remove the `... rdf:type dbpedia:Lexicon` statement from the generated RDF.

**Comment by justgo129** [comment]
Sure.

GCIS has instrument_instance(s), which is a pairing of a platform and
instrument.
e.g.:
https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager

https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager.thtml

We'll need to capture this somehow in the ontology.  Right now, we simply
are using "a prov:Entity" to describe it.

On Wed, Jul 1, 2015 at 3:54 PM, Stephan Zednik notifications@github.com
wrote:

> 1) Could you perhaps rephrase the instrument_instance issue?
> 
> 2) Having looked at the RDF for dbpedia:Lexicon (see
> http://dbpedia.org/data/Lexicon.n3), it seems that dbpedia:Lexicon is not
> a rdfs:Class or owl:Class. It is simply an RDF instance and not intended
> to be used as a type.
> 
> It is my recommendation that we remove the ... rdf:type dbpedia:Lexicon
> statement from the generated RDF.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/35#issuecomment-117806779
> .

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by zednis** [comment]
Ah, ok.  

First, is an RDF individual that describes an instrument describing a model of instrument or a specific physical object?

example - http://data.globalchange.gov/instrument/special-sensor-microwave-imager.thtml

**Comment by justgo129** [comment]
A specific physical object.  @rewolfe could you elaborate further on the characteristics of instrument instances?

**Comment by zednis** [comment]
@justgo129 do you know what template was used to generate the RDF for https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager.thtml ?  This is another case of using classes as properties and I will need to create a ticket in the gcis repo and suggest updates to the templates.

**Comment by justgo129** [comment]
https://github.com/USGCRP/gcis/blob/master/lib/Tuba/files/templates/instrument_instance/object.ttl.tut

With respect to using classes as properties, I believe that resolving the
issue labeled I1 on the aformentioned document will also resolve other
items under category I in the document.

On Wed, Jul 1, 2015 at 4:12 PM, Stephan Zednik notifications@github.com
wrote:

> @justgo129 https://github.com/justgo129 do you know what template was
> used to generate the RDF for
> https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager.thtml
> ? This is another case of using classes as properties and I will need to
> create a ticket in the gcis repo and suggest updates to the templates.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/35#issuecomment-117811939
> .

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by rewolfe** [comment]
Stephan,

Yes, an instrument/platform instance is something physical.  For example
MODIS is a type of instrument.  MODIS/Terra and MODIS/Aqua are two physical
instruments one on the Terra platform and the other on the Aqua platform.

 -Robert

On Wed, Jul 1, 2015 at 4:12 PM, Stephan Zednik notifications@github.com
wrote:

> @justgo129 https://github.com/justgo129 do you know what template was
> used to generate the RDF for
> https://data.globalchange.gov/platform/defense-meteorological-satellite-program-f-11/instrument/special-sensor-microwave-imager.thtml
> ? This is another case of using classes as properties and I will need to
> create a ticket in the gcis repo and suggest updates to the templates.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/35#issuecomment-117811939
> .

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by zednis** [comment]
@rewolfe @justgo129 

I think you misunderstood my question.  I was referring to RDF instances with type `gcis:Instrument`.  Are they resources that represent physical objects or types?

example - http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer.thtml

Based on your answers and looking at the current RDF an instance of type `gcis:Instrument` is not a physical object, but an instrument type.

**Comment by zednis** [comment]
The original request makes more sense now, because currently instances of `gcis:Instrument` do not represent physical objects but more specific _instrument types_.

We should review the domain and range of [`gcis:inPlatform`](http://data.globalchange.gov/gcis.owl#d4e558) and [`gcis:hasInstrument`](http://data.globalchange.gov/gcis.owl#d4e313) to see if they make sense.

| lay term | type | URI |
| --- | --- | --- |
| MODIS | `gcis:Instrument`,`prov:Entity` | http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer |
| MODIS/Aqua | `prov:Entity` | http://data.globalchange.gov/platform/aqua/instrument/moderate-resolution-imaging-spectroradiometer |

One option would be to create subclasses of `gcis:Instrument` for additional instrument types.  `gcis:MODIS` would be a subclass of `gcis:Instrument` and the resource representing MODIS/Aqua would have type `gcis:MODIS`.  I think this will be a pain to maintain as you will have to incorporate a large class hierarchy into your ontology.  Every time you add new instrument types to linked data you would have to update your class hierarchy - so let's not do that.  

Options:

1) rename the current `gcis:Instrument` class to `gcis:InstrumentType`.  Re-introduce the class `gcis:Instrument` as a type for physical objects.  We'll have to create a few new properties as well.

```
<http://data.globalchange.gov/platform/aqua/instrument/moderate-resolution-imaging-spectroradiometer>
   a gcis:Instrument ;
   gcis:platform <http://data.globalchange.gov/platform/aqua>;
   gcis:instrumentType <http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer> .

<http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer>
  a gcis:InstrumentType ;
 ...
```

The is similar to the RDA datatype recommendation for how to deal with a similar type-issue in datasets.

2) Leave `gcis:Instrument` as it is (perhaps updating the definition to make the intent clear) and introduce the class `gcis:InstrumentItem` (or similar) for the resources representing the physical objects.

In both options we should define a property, `gcis:instrumentType` to relate the physical object to the instrument type.

**Comment by xgmachina** [comment]
I prefer the option 1 proposed by @zednis. This is same to what we do with data types in another project. It avoids adding subclasses to `gcis:Instrument`. Users can add instances of `gcis:InstrumentType`, and can also use those instances as tags of a `gcis:Instrument` instance.

**Comment by justgo129** [comment]
Interesting discussion.  Just to ensure that I understand @zednis and @xgmachina , the proposed solution (option number 1) is as follows:

Consider what we call an "instrument_instance"
   e.g.: http://data.globalchange.gov/platform/aqua/instrument/moderate-resolution-imaging-spectroradiometer 
  as a gcis:Instrument

Consider what we call an "instrument" 
  e.g. http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer
  as a "gcis:InstrumentType"
and revising properties etc. thereof?

**Comment by zednis** [comment]
@justgo129 that is correct

**Comment by xgmachina** [comment]
+1
http://data.globalchange.gov/platform/aqua/instrument/moderate-resolution-imaging-spectroradiometer  represents the physical object of the MODIS instrument on Aqua.
http://data.globalchange.gov/instrument/moderate-resolution-imaging-spectroradiometer represents the general MODIS instrument, which is regarded as a type.

**Comment by bduggan** [comment]
This is fine, and solves the immediate problem, basically
echoing the relational representation.

At some point this should include a deeper treatment of
classification of instruments, addressing issues such as
types of types (radiometer, radar, sonar, altimeter),
modes (e.g. these are the same physical
instrument: http://data.globalchange.gov/instrument/active-microwave-instrumentation-image-mode
http://data.globalchange.gov/instrument/active-microwave-instrumentation-wave-mode),
versions (e.g. these are the same type of instrument:
http://data.globalchange.gov/instrument/doris,
http://data.globalchange.gov/instrument/doris-ng),
and different classifications of instruments vs
instrument-types by different organizations.

Brian

**Comment by aulenbac** [comment]
I agree with @bduggan

**Comment by justgo129** [comment]
Feel free to prepare the pull request, @zednis

**Comment by justgo129** [comment]
I have addressed these in https://github.com/USGCRP/gcis/pull/202 and https://github.com/USGCRP/gcis/pull/203.  However, I'll keep this ticket open as we need to update the owl file for the GCIS ontology accordingly in order to reflect these changes.

**Comment by justgo129** [comment]
Closed via merged https://github.com/USGCRP/gcis-ontology/pull/93.
